### DISCUSSION TITLE
New Transactions API

### DISCUSSION THREAD

**Discussion by jcsacto** [juftin/lunchable#115] [General]
HI @juftin , just wanted to confirm you saw the changes to the transactions API in LunchMoney coming this weekend?  I tagged a question on the Discord for you as I want to see if I should wait to deploy my solution until you respond on if this is going to break stuff.  :)  Thanks!

**Comment by juftin** [comment]
[copied from discord]

I don't actually believe these are breaking changes for lunchable - pydantic handles this quite nicely. All of the deprecated fields were already optional on the response so you will continue to see them on the TransactionObjects returned, but they'll be None. And as long as you don't use the group_id parameter on the get_transactions it won't get passed to the API.

The 35+ new fields though will take code change to expose though. They simply won't be included on the final TransactionObject that gets returned by the client. I could make a very small change to expose them as attributes automatically - but they would be undocumented, all string values by default, and you wouldn't get any code-completion in your IDE. 

Another good reason https://feedback.lunchmoney.app/developer-api/p/openapi-spec - having an OpenAPI spec could really help to create API clients dynamically.
